### PR TITLE
ci: add publish GitHub actions

### LINKED ISSUE #694: Add publish GitHub actions
We already have automatic build and release implemented with GitHub actions. We now need an automated process to publish our components.

* We decided that it's ok to store credentials using GitHub secrets.
* We will use GitHub actions to implement that.
* We need to publish in two places:
  * CDN
  * NPM
* We need to trigger publication after a release.

### PR DESCRIPTION
**Pull request by pdesoyres-cc**
Fixes #694

### CODE REVIEW

**Review by roberttran-cc — APPROVED** [review]
LGTM, well done! :tada: 

But i'll advise waiting for more (enlightened) approvals, as i do not know GitHub Actions really well. :sweat_smile:

**Review by florian-sanders-cc — APPROVED** [review]
Gg ! So glad these steps are now automated :sunglasses: 

LGTM but I'm definitely no expert :roll_eyes:

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      prerelease: ${{ steps.version_parse.outputs.prerelease }}
+      skip-publish: ${{ steps.version_parse.outputs.skip-publish }}
+    steps:
+      - name: Decode version from tag
+        id: version_regex
+        uses: KyoriPowered/action-regex-match@v3
+        with:
+          text: ${{ github.ref_name }}
+          regex: '^(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-(beta)\.(?:0|[1-9]\d*))?$'
+      - name: 'Parse version'
```
Comment (Galimede) [inline .github/workflows/publish.yml:22]: todo: Watch out you put some single quotes around the step name here where there's no quotes on every other steps or jobs name

**Review by Galimede — CHANGES_REQUESTED** [review]
GG :muscle:

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -8,13 +8,16 @@ jobs:
   release:
     name: release
     runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.release.outputs.tag_name }}
+      release_created: ${{ steps.release.outputs.release_created }}
```
Comment (roberttran-cc) [inline .github/workflows/release.yml:13]: **💣 chore:** If I understand correctly, those variables are no longer needed in this file.

Comment (pdesoyres-cc) [reply .github/workflows/release.yml:13]: exact

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        id: npm_tag
+        run: p=${{ needs.pre-publish.outputs.prerelease }} && echo "tag=${p:-latest}" >> $GITHUB_OUTPUT
+      - name: '[Prepare] Install dependencies'
+        run: npm ci
+      - name: '[Publish] version ${{ github.ref_name }} with tag ${{ steps.npm_tag.outputs.tag }}'
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: |
+          npm config set //registry.npmjs.org/:_authToken ${NPM_TOKEN}
+          npm publish --access public --tag ${{ steps.npm_tag.outputs.tag }}
```
Comment (hsablonniere) [inline .github/workflows/publish.yml:52]: praise: you really handled all the tricky details :hugs:

**Review by hsablonniere — APPROVED** [review]
Again, I'm not GitHub action expert but well done @pdesoyres-cc this is awesome!!!! :tada:

**Review by roberttran-cc — APPROVED** [review]

**Pull request merged**